app/models/user.rb

```diff
@@ -3,6 +3,8 @@ class User < ApplicationRecord
   validates :email, uniqueness: true, presence: true, case_sensitive: false
   validates :password_digest, confirmation: true
   enum role: %w(regular merchant admin)
+  # validates :merchant_id, numericality: {allow_blank: true}
```

Comment:
This will force a foreign key to a :merchant to be present whenever trying to create a user with a role of 1 (merchant)
```suggestion
  validates :merchant, presence: true, if: :merchant? 
```
(I think)